### DISCUSSION TITLE
dynamic Tags to Datadog Matrics.

### DISCUSSION THREAD

**Discussion by meetveera** [artilleryio/artillery#3337] [Artillery]
@hassy - I have been working with Artillery for a long time and I would say its a powerfull tool for Automation. But I am stuck at one point. In config section we have a fix set of what tags should be attached to a test when sending metrics to Datadog for a test run. But I am in a scenario where I am generating some data using beforeScenario hook and than using loop for each data generated. But in Datadog all the metrics are mixed for that run. But what I want is that when the data is created in beforeScenario hook and when the loop is running I want that the dynamic tag should also be created for that so that its easy to filter out the results in datadog. Is there any work around that I can use in your opinion where I can achieve this. 
Or any way where I can dynamically add tags value to a specific tags as the test are running so that I can filter out the same in datadog? 

Looking forward to hearing from you. 

Thanks

**Comment by hassy** [comment]
Hi @meetveera 👋 there's no way to customize tags on metrics sent by Artillery to Datadog. All metrics share the same tags which are set on startup. Are you emitting custom metrics from your `beforeScenario` hook? If you are, you could possibly use a different name that you could filter on?
